### PR TITLE
[Core] Refactor EnsureModelPartOwnsPropertie.Making available from core

### PR DESCRIPTION
**Pull request by loumalouomega**
Old methods are preserved for backward-compatibility

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
Ping @KratosMultiphysics/technical-committee

**Comment by philbucher** [comment]
so this is some sort of cleanup?

looks ok to me, but I would like someone else to also take a look

**Comment by loumalouomega** [comment]
> 
> 
> so this is some sort of cleanup?
> 
> looks ok to me, but I would like someone else to also take a look

Yes, also in order to be able to use it without needing to compile the meshing

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 1]
```diff
@@ -12,14 +12,94 @@
...
+    auto& r_conditions_array = mrModelPart.Conditions();
+    const auto it_cond_begin= r_conditions_array.begin();
+    const int number_of_conditions = static_cast<int>(r_conditions_array.size());
+
+    #pragma omp parallel
+    {
+        // The list of properties
+        std::unordered_set<Properties::Pointer, IndexedObjecPointertHasher<Properties::Pointer>, IndexedObjectPointerComparator<Properties::Pointer>> buffer_list_of_properties;
+
+        #pragma omp for schedule(guided, 512) nowait
```
Comment (pooyan-dadvand) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]: why guided and not static (or even dynamic)?

**Review by pooyan-dadvand — COMMENTED** [review]
Only a small comment on OpenMP. 

Meanwhile, I would never have found it in the in auxiliar_model_part_utilities.h. Why is not model_part_utilities?

**Comment by loumalouomega** [comment]
> Meanwhile, I would never have found it in the in auxiliar_model_part_utilities.h. Why is not model_part_utilities?

@RiccardoRossi suggested that name

File: kratos/utilities/auxiliar_model_part_utilities.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/utilities/auxiliar_model_part_utilities.cpp]: Certainly I think is copy paste from other side, I can change it

**Comment by pooyan-dadvand** [comment]
Would you please add some test?

**Comment by loumalouomega** [comment]
> 
> 
> Would you please add some test?

Done @pooyan-dadvand

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
This is ready, any further comment? @KratosMultiphysics/technical-committee

**Review by pooyan-dadvand — APPROVED** [review]
Thanks!

**Pull request merged**